companion-api/api/user.go

```diff
@@ -5,43 +5,36 @@ import (
 
 	restful "github.com/emicklei/go-restful"
 	"github.com/mcluseau/autorizo/api"
-	"github.com/mcluseau/autorizo/auth"
 	"github.com/mcluseau/autorizo/companion-api/backend"
 )
 
+// CreateUserReq is a request to create a new UserData
+type CreateUserReq struct {
+	Id   string           `json:"id"`
+	User backend.UserData `json:"user"`
+}
+
+// UpdateUserReq is a request to modify a existing UserData
+type UpdateUserReq struct {
+	Id     string                  `json:"id"`
+	Patchs []backend.UserDataPatch `json:"patchs"`
+}
+
 // Register provide a restful.WebService from this API
 func (cApi *CompanionAPI) registerUsers(ws *restful.WebService) {
 	ws.
-		Route(ws.POST("/user/{user-id}").
+		Route(ws.POST("/users").
 			To(cApi.createUser).
 			Doc("Create a new user.").
 			Consumes("application/json").
-			Param(ws.PathParameter("user-id", "identifier of the user").DataType("string")).
-			Reads(backend.UserData{}))
+			Reads(CreateUserReq{}))
 
 	ws.
-		Route(ws.PUT("/user/{user-id}").
+		Route(ws.PUT("/users").
 			To(cApi.updateUser).
 			Doc("Update password of an existing user.").
 			Consumes("application/json").
-			Param(ws.PathParameter("user-id", "identifier of the user").DataType("string")).
-			Reads(backend.UserData{}))
-
-	ws.
-		Route(ws.PUT("/user/{user-id}/password").
-			To(cApi.updateUserPassword).
-			Doc("Update password of an existing user.").
-			Consumes("application/json").
-			Param(ws.PathParameter("user-id", "identifier of the user").DataType("string")).
-			Reads(backend.PasswordHashData{}))
-
-	ws.
-		Route(ws.PUT("/user/{user-id}/claims").
-			To(cApi.updateUserClaims).
-			Doc("Update claims of an existing user.").
-			Consumes("application/json").
-			Param(ws.PathParameter("user-id", "identifier of the user").DataType("string")).
-			Reads(auth.ExtraClaims{}))
+			Reads(UpdateUserReq{}))
```

Comment:
`PUT` is to set a new value, so it reads `UserData`. Patching is done with the HTTP verb... `PATCH` ;)
https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PATCH